source/asdmanager.py

```diff
@@ -110,23 +113,39 @@ def setup():
         print Interactive.boxed_message(['API port cannot be in the range of the ASD port + 100'])
         sys.exit(1)
 
-    print '- Initializing etcd'
+    store = Interactive.ask_choice(['Arakoon', 'Etcd'],
+                                   question='Select the configuration management system',
+                                   default_value='Arakoon').lower()
+    if store == 'arakoon':
+        from source.tools.configuration.arakoon_config import ArakoonConfiguration
+        file_location = ArakoonConfiguration.CACC_LOCATION
+        while not local_client.file_exists(file_location):
+            print 'Please place a copy of the Arakoon\'s client configuration file at: {0}'.format(file_location)
+            Interactive.ask_continue()
+    bootstrap_location = Configuration.BOOTSTRAP_CONFIG_LOCATION
+    if not local_client.file_exists(bootstrap_location):
+        local_client.file_create(bootstrap_location)
+    local_client.file_write(bootstrap_location, json.dumps({'configuration_store': store}, indent=4))
     try:
-        alba_node_id = EtcdConfiguration.initialize(api_ip, api_port, asd_ips, asd_start_port)
+        alba_node_id = Configuration.initialize(api_ip, api_port, asd_ips, asd_start_port)
     except:
-        print ''  # Spacing
-        print Interactive.boxed_message(['Could not connect to Etcd.',
-                                         'Please make sure an Etcd proxy is available, pointing towards an OpenvStorage cluster.'])
+        print ''
+        if store == 'arakoon':
+            print Interactive.boxed_message(['Could not connect to Arakoon'])
+        else:
+            print Interactive.boxed_message(['Could not connect to Etcd.',
+                                             'Please make sure an Etcd proxy is available, pointing towards an OpenvStorage cluster.'])
         sys.exit(1)
 
     ServiceManager.add_service(service_name, local_client, params={'ASD_NODE_ID': alba_node_id,
                                                                    'PORT_NUMBER': str(api_port)})
-    print '- Starting ASD manager service'
+    ServiceManager.add_service(watcher_name, local_client)
+    print '- Starting watcher service'
     try:
-        ServiceManager.start_service(service_name, local_client)
+        ServiceManager.start_service(watcher_name, local_client)
     except Exception as ex:
-        EtcdConfiguration.uninitialize(alba_node_id)
-        print Interactive.boxed_message(['Starting asd-manager failed with error:', str(ex)])
+        dConfiguration.uninitialize(alba_node_id)
```

Comment:
I hope we'll never run into this exception, cause i'm afraid we'll see a different stacktrace than expected